paperqa/agents/main.py

```diff
@@ -120,7 +125,7 @@ async def run_agent(
     # Build the index once here, and then all tools won't need to rebuild it
     # only build if the a search tool is requested
     if PaperSearch.TOOL_FN_NAME in (settings.agent.tool_names or DEFAULT_TOOL_NAMES):
-        await get_directory_index(settings=settings)
+        await get_directory_index(settings=settings, build=force_index_rebuild)
```

Comment:
If I understand this change, basically you're calling `agent_query ` multiple times with the same `IndexSettings.name`. And are finding it will rebuild the index every time?

We made this `agent_query` flow for CLI users, hence why there is the colorized logging. Internally, we just directly work with the core `PaperQAEnvironment`.

That being said, I think this change makes sense, but with some tweaks

Let's do this:

1. Make a new setting in `AgentSettings` called `rebuild_index`, defaulted to `True`, with a `description`
2. Access that setting here, instead of adding keyword arguments to `run_agent`/`agent_query`

Reply:
> If I understand this change, basically you're calling agent_query  multiple times with the same IndexSettings.name. And are finding it will rebuild the index every time?

Yes this is exactly what I am doing. And this is exactly the issue.

>  Internally, we just directly work with the core PaperQAEnvironment

Is there also documentation for that?

> Let's do this:
> 
> Make a new setting in AgentSettings called rebuild_index, defaulted to True, with a description
> Access that setting here, instead of adding keyword arguments to run_agent/agent_query

That sounds good, I will update the PR tomorrow, and also include a regex in the test.

Reply:
> Is there also documentation for that?

Yeah PaperQA's docs are quite lacking, there's not nearly enough. In our current docs, probably in https://github.com/Future-House/paper-qa#running-on-litqa-v2. It uses an environments framework designed for training agents to use it called aviary ([paper](https://arxiv.org/abs/2412.21154), [code](https://github.com/Future-House/aviary)), if you are curious.

Feel free to just ignore this part, I think use of `run_agent` or `agent_query` is quite fine too.